pkg/raft/node.go

```diff
@@ -417,18 +440,19 @@ func (n *node) run() {
 			}
 		case <-n.tickc:
 			n.rn.Tick()
-		case readyc <- rd:
-			n.rn.acceptReady(rd)
-			if !n.rn.asyncStorageWrites {
-				advancec = n.advancec
-			} else {
+		case acceptReadyC <- struct{}{}:
+			// Invariant: n.rn.HasReady() == true.
+			rd = n.rn.Ready()
+			n.readyW.resp <- rd // never blocks
+			if n.rn.asyncStorageWrites {
 				rd = Ready{}
+			} else {
+				n.readyW.needAdvance = true
 			}
-			readyc = nil
-		case <-advancec:
+		case <-n.readyW.advC():
 			n.rn.Advance(rd)
 			rd = Ready{}
```

Comment:
I know this was there before but still, is there a reason for zeroing `rd` here and in the `advC()` case?

Reply:
I guess it's just not to reference any memory since this `Ready` is no longer used. And it might take a while before it's rewritten to a new `Ready`.